### PR TITLE
WIP for comments -  test_Client and test_cloudflare pass, expect othe…

### PR DESCRIPTION
**Pull request by mmaney**
…rs to fail

See #157 - this is a minimal (!) start on it, with some tests passing.

Moved dns_providers that don't have deps into auth_providers, similarly their tests.  Only cloudflare has been patched (and OMG did the automated "clean" formatting of that stack of mock.patches make it impossible to read, which is why I expect black to complain about this, but that's irrelevant).

Oh, I forgot, meant to patch readme.md to see how little this actually changes the publicly visible advice about how to use sewer as an API.

### CODE REVIEW

**Comment by komuw** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- sewer/auth_providers/common.py  1
- sewer/auth_providers/tests/test_utils.py  4
- sewer/auth_providers/__init__.py  1
- sewer/auth_providers/tests/test_cloudflare.py  5
         

Clones added
============
- sewer/auth_providers/common.py  1
- sewer/auth_providers/tests/test_utils.py  1
- sewer/auth_providers/tests/test_cloudflare.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/komuW/sewer/pullRequest?prid=5264928&bid=17247572)

File: sewer/auth_providers/tests/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+class mockLibcloudDriver(object):
+    """
+    a mock of libcloud.dns.drivers.auroradns.AuroraDNSDriver class
+    """
+
+    def __init__(self, key, secret):
+        pass
+
+    def get_zone(self, domainSuffix):
```
Comment (komuw) [inline sewer/auth_providers/tests/test_utils.py]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/komuW/sewer/pullRequest?prid=5264928)

File: sewer/auth_providers/tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+
+class mockLibcloudDriverZone(object):
+    """
+    A mock of a dns zone in a libcloud drivers dns
+    """
+
+    id = "mock-zone-id-1"
+
+    def create_record(self, name, type, data):
```
Comment (komuw) [inline sewer/auth_providers/tests/test_utils.py]: Codacy Issue found: [Redefining built-in 'type'](https://app.codacy.com/manual/komuW/sewer/pullRequest?prid=5264928)

File: sewer/auth_providers/tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    def list_records(self, zone):
+        import collections
+
+        DnsRecords = collections.namedtuple("DnsRecords", "id name type")
+        one_dns_record = DnsRecords(id="1", name="_acme-challenge", type="TXT")
+        records = [one_dns_record]
+        return records
+
+    def get_record(self, zone_id, record_id):
```
Comment (komuw) [inline sewer/auth_providers/tests/test_utils.py]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/komuW/sewer/pullRequest?prid=5264928)

File: sewer/auth_providers/tests/test_utils.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    """
+
+    def __init__(self, key, secret):
+        pass
+
+    def get_zone(self, domainSuffix):
+        mock_zone = mockLibcloudDriverZone()
+        return mock_zone
+
+    def list_records(self, zone):
```
Comment (komuw) [inline sewer/auth_providers/tests/test_utils.py]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/komuW/sewer/pullRequest?prid=5264928)

File: sewer/auth_providers/common.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        record:
+        """
+        domain_name = identifier_auth["domain"]
+        base64_of_acme_keyauthorization = calculate_safe_base64(
+            sha256(acme_keyauthorization.encode("utf8")).digest()
+        )
+        self.create_dns_record(domain_name, base64_of_acme_keyauthorization)
+        return {"domain_name": domain_name, "value": base64_of_acme_keyauthorization}
+
+    def cleanup_authorization(self, domain_name, value):
```
Comment (komuw) [inline sewer/auth_providers/common.py]: Codacy Issue found: [Parameters differ from overridden 'cleanup_authorization' method](https://app.codacy.com/manual/komuW/sewer/pullRequest?prid=5264928)

File: sewer/auth_providers/tests/test_utils.py (hunk 1 above)
Comment (mmaney) [reply sewer/auth_providers/tests/test_utils.py]: Sorry, that came with test_utils when I copied it into auth_providers.  Not my fault!  :-)

**Comment by mmaney** [comment]
We'll see this again, in somewhat revised form.

**Pull request closed without merging**